base/int.jl

```diff
@@ -376,46 +376,22 @@ if WORD_SIZE == 32
     end
 
     function div(x::Int128, y::Int128)
-        try
-            Int128(div(BigInt(x),BigInt(y)))
-        catch e
-            isa(e, InexactError) && throw(DivideError())
-            rethrow()
-        end
+        (x == typemin(Int128)) & (y == -1) && throw(DivideError())
```

Comment:
A comparison with -1 is very cheap, and the generated native code is simpler (and probably more efficient) at least on x86-64 when I use `&` instead of `&&`.

I humbly think that, in this day and age, LLVM should really be able to figure out that both are equivalent, and should choose the most efficient code independent of whether I write `&` or `&&`. Alas, that's not the case yet.